hedera-node/src/main/java/com/hedera/services/store/contracts/HederaWorldState.java

```diff
@@ -69,104 +69,41 @@
-			}
-		});
-
-		final var createdContracts = new ArrayList<ContractID>();
-
-		// copy over sponsor account info for CREATE operations
-		sponsorMap.forEach((contract, sponsorAddress) -> {
-			if (!provisionalAccountUpdates.containsKey(contract)) return;
-
-			AccountID sponsorAccount = EntityIdUtils.accountParsedFromSolidityAddress(sponsorAddress.toArrayUnsafe());
-
-			WorldStateAccount provisionalAccount = provisionalAccountUpdates.get(contract);
-			if (ledger.exists(sponsorAccount)) {
-				MerkleAccount sponsorMerkleAccount = ledger.get(sponsorAccount);
-				provisionalAccount.setProxyAccount(sponsorMerkleAccount.getProxy());
-				provisionalAccount.setKey(sponsorMerkleAccount.getAccountKey());
-				provisionalAccount.setMemo(sponsorMerkleAccount.getMemo());
-				provisionalAccount.setAutoRenew(sponsorMerkleAccount.getAutoRenewSecs());
-				provisionalAccount.setExpiry(sponsorMerkleAccount.getExpiry());
-			} else if (provisionalAccountUpdates.containsKey(sponsorAddress)) {
-				WorldStateAccount sponsorProvisionalAccount = provisionalAccountUpdates.get(sponsorAddress);
-				provisionalAccount.setProxyAccount(sponsorProvisionalAccount.getProxyAccount());
-				provisionalAccount.setKey(sponsorProvisionalAccount.getKey());
-				provisionalAccount.setMemo(sponsorProvisionalAccount.getMemo());
-				provisionalAccount.setAutoRenew(sponsorProvisionalAccount.getAutoRenew());
-				provisionalAccount.setExpiry(sponsorProvisionalAccount.getExpiry());
-			}
-		});
-
-		provisionalAccountUpdates.forEach((address, worldStateAccount) -> {
-			final var accountId = accountParsedFromSolidityAddress(address.toArray());
-			if (ledger.exists(accountId)) {
-				long oldBalance = ledger.getBalance(accountId);
-				long newBalance = worldStateAccount.getBalance().toLong();
-				long adjustment = newBalance - oldBalance;
-				ledger.adjustBalance(accountId, adjustment);
-			} else {
-//				var key = new JContractIDKey(worldStateAccount.getProxyAccount().toGrpcContractId()); // TODO: might be used for factory contracts
-				HederaAccountCustomizer customizer = new HederaAccountCustomizer()
-						.key(worldStateAccount.getKey())
-						.memo("")
-						.proxy(worldStateAccount.getProxyAccount())
-						.expiry(worldStateAccount.getExpiry())
-						.autoRenewPeriod(worldStateAccount.getAutoRenew())
-						.isSmartContract(true);
-				if (provisionalAccountDeletes.contains(address)) {
-					customizer.isDeleted(true);
-				}
-
-				long balance = worldStateAccount.getBalance().toLong();
-				ledger.spawn(accountId, balance, customizer);
-				createdContracts.add(EntityIdUtils.asContract(accountId));
-			}
-		});
+	public List<ContractID> persist(HederaAccountCustomizer accountCustomizer) {
 
 		/* Commit code updates for each updated address */
-		provisionalCodeUpdates.forEach((address, code) -> repositoryRoot.saveCode(address.toArray(), code.toArray()));
-
 		repositoryRoot.flush();
 
+		if (accountCustomizer != null) {
```

Comment:
I think that the customizer should not be `null` if the transaction is not successful. We could provide it always. The list of `createdContracts` will be empty so nothing will happen. This way we will have a more clean Transition Logics as-well